app/controllers/groups_controller.rb

```diff
@@ -38,52 +38,103 @@ class GroupsController < ApplicationController
   before_action :find_group, only: %i[destroy show create_memberships destroy_membership
                                       edit_membership add_users]
 
+  # GET /groups
+  # GET /groups.xml
   def index
     @groups = Group.order(Arel.sql('lastname ASC'))
+
+    respond_to do |format|
+      format.html # index.html.erb
+      format.xml { render xml: @groups }
+    end
   end
 
+  # GET /groups/1
+  # GET /groups/1.xml
   def show
-    @group_users = group_members
-    render layout: 'no_menu'
+    respond_to do |format|
+      format.html do
+        @group_users = group_members
+        render layout: 'no_menu'
+      end
+      format.xml { render xml: @group }
+    end
   end
 
+  # GET /groups/new
+  # GET /groups/new.xml
   def new
     @group = Group.new
+
+    respond_to do |format|
+      format.html # new.html.erb
+      format.xml  { render xml: @group }
+    end
   end
 
+  # GET /groups/1/edit
   def edit
     @group = Group.includes(:members, :users).find(params[:id])
 
     set_filters_for_user_autocompleter
   end
 
+  # POST /groups
+  # POST /groups.xml
   def create
     @group = Group.new permitted_params.group
 
-    if @group.save
-      flash[:notice] = I18n.t(:notice_successful_create)
-      redirect_to(groups_path)
-    else
-      render action: :new
+    respond_to do |format|
+      if @group.save
+        flash[:notice] = I18n.t(:notice_successful_create)
+        format.html { redirect_to(groups_path) }
+        format.xml  { render xml: @group, status: :created, location: @group }
+      else
+        format.html { render action: :new }
+        format.xml  { render xml: @group.errors, status: :unprocessable_entity }
+      end
+      if @group.save
+        flash[:notice] = I18n.t(:notice_successful_create)
+        redirect_to(groups_path)
+      else
+        render action: :new
+      end
```

Comment:
There are two blocks on `@group.save` which results in double rendering. Probably simply remove the second group.